c7n/policy.py

```diff
@@ -470,6 +471,15 @@ def run(self, event, lambda_context):
                 self.policy._write_file(
                     "action-%s" % action.name, utils.dumps(results))
         return resources
+    def expand_variables(self, policy):
+        """expand any variables in the action to_from/cc_from fields.
+        """
+        p = self.policy.data.copy()
+        if 'mode' in self.policy.data:
+            mode = self.policy.data['mode'].copy()
+            mode['role'] = re.sub(r'{{account_id}}',policy['account_id'],mode['role'])
```

Comment:
any reason for the sub vs string format here? also need to check conditionally check if role is specified, we typically use --assume on the cli to pick this up for cross account usage.